src/resultset/formatters/resultset_replycompact.c

```diff
@@ -175,6 +180,7 @@ void ResultSet_ReplyWithCompactHeader(RedisModuleCtx *ctx, const char **columns,
 				t = COLUMN_RELATION;
 				break;
 			case REC_TYPE_SCALAR:
+			case REC_TYPE_UNKNOWN:  // Treat unknown as scalar.
```

Comment:
Yes, occurs when we have a query that returns no results:
`MATCH (a)-[e:doesnt_exist]->(b) RETURN *`

Our result set looks like:
```
1) 1) "a"
   2) "e"
   3) "b"
2) (empty list or set)
3) 1) "Query internal execution time: 0.549402 milliseconds"
```
The types associated with the header fields don't matter since there are no records to process.